agent/lib/broker_address_settings.js

```diff
@@ -86,7 +146,7 @@ BrokerAddressSettings.prototype.delete_settings = function (plan) {
 };
 
 BrokerAddressSettings.prototype.updated = function (objects) {
-    var plans = objects.map(extract_address_plan).filter(required_broker_resource);
+    var plans = objects.filter((plan) => plan !== undefined).map(extract_spec).filter(required_broker_resource);
```

Comment:
is it really necessary to filter for undefined objects? sounds fishy to me.

Reply:
was able to remove the filter